Zend/zend_smart_str.c

```diff
@@ -38,7 +38,10 @@ ZEND_API void ZEND_FASTCALL smart_str_erealloc(smart_str *str, size_t len)
 		ZSTR_LEN(str->s) = 0;
 	} else {
 		str->a = SMART_STR_NEW_LEN(len);
-		str->s = (zend_string *) erealloc2(str->s, str->a + _ZSTR_HEADER_SIZE + 1, _ZSTR_HEADER_SIZE + ZSTR_LEN(str->s));
+	    void *temp = erealloc2(str->s, str->a + _ZSTR_HEADER_SIZE + 1, _ZSTR_HEADER_SIZE + ZSTR_LEN(str->s));
```

Comment:
`erealloc` is an infallible allocator, so the return value does not need to be checked.